File: solutions/secret-handshake/Python/secret_handshake.py
actions = {
    0b0001 : "wink",
    0b0010 : "double blink",
    0b0100 : "close your eyes",
    0b1000 : "jump"
}

def commands(binary_str):
    input = int(binary_str, 2)
    code = [actions[i] for i in actions if i & input]
    if input & 0b10000: code.reverse()
    return code

File: solutions/secret-handshake/C++/secret_handshake.cpp
#include "secret_handshake.h"

namespace secret_handshake {
    using namespace std;
    std::vector<std::string> commands(unsigned int number) {
        std::vector<std::string> handshake {};
        if (number & (1U << 0)) handshake.emplace_back("wink");
        if (number & (1U << 1)) handshake.emplace_back("double blink");
        if (number & (1U << 2)) handshake.emplace_back("close your eyes");
        if (number & (1U << 3)) handshake.emplace_back("jump");
        if (number & (1U << 4)) std::reverse(handshake.begin(), handshake.end());
        return handshake;  
    }
}
